src/utils/legacyLayerParamUtils.js

```diff
@@ -123,9 +126,28 @@ export function getLayersFromLegacyUrlParams(layersConfig, legacyLayersParam) {
                         }
                         // checking if a timestamp is defined for this layer
                         if (layerTimestamps.length > index && layerTimestamps[index]) {
-                            layer.timeConfig.updateCurrentTimeEntry(
+                            let year = layerTimestamps[index]
+                            if (
                                 layer.timeConfig.getTimeEntryForTimestamp(layerTimestamps[index])
-                            )
+                                    .year
+                            ) {
+                                year = layer.timeConfig.getTimeEntryForTimestamp(
+                                    layerTimestamps[index]
+                                ).year
+                            }
+                            if (store) {
+                                store.dispatch('setLegacyTimeLayerYear', {
+                                    layer: layer,
+                                    year: year,
+                                })
+                            } else {
+                                // this does tell us that the changes have been taken into account
+                                // and would be updated, without using a store that is not active
+                                // during unit tests
+                                layer.timeConfig.updateCurrentTimeEntry(
+                                    layer.timeConfig.getTimeEntryForYear(year)
+                                )
+                            }
```

Comment:
`layer.timeConfig.updateCurrentTimeEntry` creates a vue Error at runtime, telling us it can't modify the store outside of a mutation

the `if(store) ... else` part is quite hacky indeed, as the unit tests crashed, the store not being instantiated during unit tests.

I'll try to see if I can avoid the store completely during this. I feel like, with these layers being cloned to be modified before pushing them to be activated, we should be able to do so, but I failed to make it work.

Reply:
This should not, you should be able to use the `updateCurrentTimeEntry` on a layer that has been cloned you probably had another issue in your code.
Here you are in the context of a router plugin that will update the store with a new layer object, so you don't have to do any store dispatch.

Reply:
You might need to check that the layer clone method is cloning also the layer config this might be an issue and the reason of the vue failure